AmplifyTests/CategoryTests/Auth/AuthCategoryConfigurationTests.swift

```diff
@@ -0,0 +1,367 @@
+                             "Getting a plugin after reset() should throw") { error in
+                                guard case AmplifyAuthError.configuration = error else {
+                                    XCTFail("Expected PluginError.noSuchPlugin")
+                                    return
+                                }
+        }
+    }
+
+    /// Test if we can register multiple plugins
+    ///
+    /// - Given: UnConfigured Amplify framework
+    /// - When:
+    ///    - I configure Amplify with multiple plugins for Auth
+    /// - Then:
+    ///    - I should be able to access individual plugins I added.
+    ///
+    func testCanRegisterMultiplePlugins() throws {
+        let plugin1 = MockAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin1)
+
+        let plugin2 = MockSecondAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin2)
+
+        let config = AuthCategoryConfiguration(
+            plugins: [
+                "MockAuthCategoryPlugin": true,
+                "MockSecondAuthCategoryPlugin": true
+            ]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+
+        XCTAssertNotNil(try Amplify.Auth.getPlugin(for: "MockAuthCategoryPlugin"))
+        XCTAssertNotNil(try Amplify.Auth.getPlugin(for: "MockSecondAuthCategoryPlugin"))
+    }
+
+    /// Test if the default plugin works
+    ///
+    /// - Given: Amplify configured with Auth plugin
+    /// - When:
+    ///    - I invoke a API from Auth plugin with default invocation
+    /// - Then:
+    ///    - API should complete without error
+    ///
+    func testCanUseDefaultPluginIfOnlyOnePlugin() throws {
+        let plugin = MockAuthCategoryPlugin()
+        let methodInvokedOnDefaultPlugin = expectation(description: "test method invoked on default plugin")
+        plugin.listeners.append { message in
+            if message == "changePassword" {
+                methodInvokedOnDefaultPlugin.fulfill()
+            }
+        }
+        try Amplify.add(plugin: plugin)
+
+        let config = AuthCategoryConfiguration(plugins: ["MockAuthCategoryPlugin": true])
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+
+        _ = Amplify.Auth.changePassword(currentPassword: "current",
+                                        newPassword: "new",
+                                        listener: nil)
+        waitForExpectations(timeout: 1.0)
+    }
+
+    /// Test if I can pick a specific plugin
+    ///
+    /// - Given: Amplify configured with multiple Auth plugins
+    /// - When:
+    ///    - I coose one plugin and call one of the Auth API
+    /// - Then:
+    ///    - API should complete without error for one plugin
+    ///
+    func testCanUseSpecifiedPlugin() throws {
+        let plugin1 = MockAuthCategoryPlugin()
+        let methodShouldNotBeInvokedOnDefaultPlugin =
```

Comment:
plugin1 and defaultPlugin seem to be synonymous -- seems like we should pick a name and just use one name.